libraries/AP_NavEKF3/AP_NavEKF3_Control.cpp

```diff
@@ -473,25 +473,29 @@ bool NavEKF3_core::setOriginLLH(const Location &loc)
     EKF_origin = loc;
     ekfGpsRefHgt = (double)0.01 * (double)EKF_origin.alt;
     // define Earth rotation vector in the NED navigation frame at the origin
-    calcEarthRateNED(earthRateNED, _ahrs->get_home().lat);
+    calcEarthRateNED(earthRateNED, loc.lat);
     validOrigin = true;
     return true;
 }
 
 // Set the NED origin to be used until the next filter reset
-void NavEKF3_core::setOrigin()
+void NavEKF3_core::setOrigin(const Location &loc)
 {
     // assume origin at current GPS location (no averaging)
```

Comment:
Same comment as per EKF2. This function no longer assumes a GPS origin source.

Reply:
fixed